src/miral/simulated_secondary_click_config.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "miral/simulated_secondary_click_config.h"
+
+#include "mir/log.h"
+#include "mir/options/option.h"
+#include "mir/server.h"
+#include "mir/shell/accessibility_manager.h"
+
+struct miral::SimulatedSecondaryClickConfig::Self
+{
+    Self(bool enabled_by_default) :
+        enabled_by_default{enabled_by_default}
+    {
+    }
+
+    std::weak_ptr<mir::shell::AccessibilityManager> accessibility_manager;
+    bool const enabled_by_default;
+};
+
+miral::SimulatedSecondaryClickConfig::SimulatedSecondaryClickConfig(bool enabled_by_default) :
+    self{std::make_shared<Self>(enabled_by_default)}
+{
+}
+
+void miral::SimulatedSecondaryClickConfig::enabled(bool enabled) const
+{
+    if (auto const accessibility_manager = self->accessibility_manager.lock())
+    {
+        self->accessibility_manager.lock()->simulated_secondary_click_enabled(enabled);
+    }
+    else
+    {
+        mir::log_error("AccessibilityManager not initialized. Will not toggle simulated secondary click");
+        return;
+    }
```

Comment:
Wouldn't it make more sense to the shell if we recorded the setting for use when the init callback runs?